### PR TITLE
feat: GitHub repo to profile

### LINKED ISSUE #2835: [FEATURE] Separate sections for projects
### A separate section to showcase the projects.
Currently, one can add links of their projects in `My Links` section through `links` field in their json data.

I think `My Links` is a more general space, one can figure out what to put there.

But having a `Projects` section can facilitate showcasing projects that one have worked on

### The data can look like
```json
 
    "projects": [
        {
            "title": "The name of the project",
            "techsUsed": [
                "ReactJS",
                "Typescript"
            ],
            "url": "url to the project if deployed",
            "githubRepo": " url to the repo if exists ",
            "contributors": []
        }
    ]

```

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by eddiejaoude**
## Fixes Issue

fixes #2835 

- [x] User can add/delete their GitHub repo in their profile
- [x] Repo data is fetched from GitHub's API
- [x] Display new tab `Project` on profile
- [x] GitHub data is cached (similar to GitHub profile location data)
- [x] When to update GitHub data
- [ ] ~~Have overall app page for projects (similar to Events)~~ after more projects are added

## Check List (Check all the applicable boxes) 

- [ ] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] This PR does not contain plagiarized content.
- [ ] The title of my pull request is a short description of the requested changes.

## Screenshots

Screenshot 2023-07-29 at 09 51 58
Screenshot 2023-07-29 at 09 51 53

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/8225"></a>

### CODE REVIEW

File: pages/api/account/manage/repo/[[...data]].js [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+  const log = logger.child({ username });
+  let getRepo = {};
+
+  try {
+    await Repo.validate(addRepo, ["url"]);
+  } catch (e) {
+    log.error(e, `validation failed to add repo for username: ${username}`);
+    return { error: e.errors };
+  }
+  const githubData = await getGitHubRepo(addRepo.url);
```
Comment (eddiejaoude) [inline pages/api/account/manage/repo/[[...data]].js:86]: If there is an error return early

**Comment by schmelto** [comment]
@eddiejaoude how do you can delete a repo from the profile? It seems to me that only the Post method is accessible in the website.

**Comment by eddiejaoude** [comment]
Yes that is correct, I will add delete functionality next

**Comment by eddiejaoude** [comment]
Documentation 

Screenshot 2023-08-01 at 17 54 54


Screenshot 2023-08-01 at 17 57 34

Screenshot 2023-08-01 at 18 13 52

**Pull request merged**